sdk/auth/token.go

```diff
@@ -26,7 +65,11 @@ func tokenDueForRenewal(token *oauth2.Token) bool {
 	expiry := token.Expiry.Round(0)
 	delta := tokenExpiryDelta
 	now := time.Now()
-	expiresWithinTenMinutes := expiry.Add(-delta).Before(now)
+
+	// Always return early if the token validity doesn't extend past the expiry delta
+	if expiry.Add(-delta).Before(now) {
```

Comment:
There's a `sub`tract we can use here:

```suggestion
	if expiry.Sub(delta).Before(now) {
```

Reply:
`Sub` actually does the inverse and subtracts two `Time` structs to return a `Duration`